### ISSUE TITLE
Updates to the product attribute property "apply_to" do not work via REST API

### ISSUE THREAD

**Issue by magolovanov** [magento/magento2#20629]
### Preconditions (*)

1. Magento 2.2.5
2. Amazon Linux 2 (https://aws.amazon.com/amazon-linux-ami/2018.03-release-notes/)
3. PHP 7.0.31 (cli) (built: Aug 17 2018 22:04:06) ( NTS )
4. MySQL-5.6.41

### Steps to reproduce (*)
1. Execute "manufacturer" attribute update via the API (PUT "/rest/V1/products/attributes/manufacturer") with the body

{"attribute":{"default_frontend_label":"Manufacturer updated","attribute_id":83,"apply_to":["simple","grouped"],"attribute_code":"manufacturer"}}

2. Execute "manufacturer" attribute update via the API (POST "/rest/V1/products/attributes") with the body

{"attribute":{"default_frontend_label":"Manufacturer","attribute_id":83,"apply_to":["simple","grouped"],"attribute_code":"manufacturer"}}

### Expected result (*)
1. "manufacturer" data After PUT request above:

```
{
  "is_wysiwyg_enabled": false,
  "is_html_allowed_on_front": false,
  "used_for_sort_by": false,
  "is_filterable": true,
  "is_filterable_in_search": false,
  "is_used_in_grid": true,
  "is_visible_in_grid": false,
  "is_filterable_in_grid": true,
  "position": 0,
  "apply_to": [
    "simple",
    "grouped"
  ],
  "is_searchable": "1",
  "is_visible_in_advanced_search": "1",
  "is_comparable": "1",
  "is_used_for_promo_rules": "0",
  "is_visible_on_front": "0",
  "used_in_product_listing": "0",
  "is_visible": true,
  "scope": "global",
  "attribute_id": 83,
  "attribute_code": "manufacturer",
  "frontend_input": "select",
  "entity_type_id": "4",
  "is_required": false,
  "options": [
    {
      "label": " ",
      "value": ""
    },
    {
      "label": "AMD",
      "value": "5"
    },
    {
      "label": "Apple",
      "value": "11"
    },
    {
      "label": "Argus",
      "value": "21"
    },
    {
      "label": "At&t",
      "value": "18"
    },
    {
      "label": "BlackBerry",
      "value": "16"
    },
    {
      "label": "Canon",
      "value": "20"
    },
    {
      "label": "Crucial",
      "value": "4"
    },
    {
      "label": "HTC",
      "value": "15"
    },
    {
      "label": "Intel",
      "value": "8"
    },
    {
      "label": "Kenneth Cole",
      "value": "19"
    },
    {
      "label": "Kodak",
      "value": "23"
    },
    {
      "label": "LG",
      "value": "9"
    },
    {
      "label": "Logitech",
      "value": "14"
    },
    {
      "label": "Microsoft",
      "value": "12"
    },
    {
      "label": "Nokia",
      "value": "13"
    },
    {
      "label": "Olympus",
      "value": "22"
    },
    {
      "label": "Samsung",
      "value": "6"
    },
    {
      "label": "Seagate",
      "value": "10"
    },
    {
      "label": "Sony",
      "value": "17"
    },
    {
      "label": "Western Digital",
      "value": "7"
    }
  ],
  "is_user_defined": true,
  "default_frontend_label": "Manufacturer updated",
  "frontend_labels": null,
  "backend_type": "int",
  "source_model": "Magento\\Eav\\Model\\Entity\\Attribute\\Source\\Table",
  "default_value": "",
  "is_unique": "0",
  "validation_rules": [
  ]
}
```

2. "manufacturer" data After POST request above:

{"is_wysiwyg_enabled":false,"is_html_allowed_on_front":false,"used_for_sort_by":false,"is_filterable":true,"is_filterable_in_search":false,"is_used_in_grid":true,"is_visible_in_grid":false,"is_filterable_in_grid":true,"position":0,**"apply_to":["simple","grouped"]**,"is_searchable":"1","is_visible_in_advanced_search":"1","is_comparable":"1","is_used_for_promo_rules":"0","is_visible_on_front":"0","used_in_product_listing":"0","is_visible":true,"scope":"global","attribute_id":83,"attribute_code":"manufacturer","frontend_input":"select","entity_type_id":"4","is_required":false,"options":[{"label":" ","value":""},{"label":"AMD","value":"5"},{"label":"Apple","value":"11"},{"label":"Argus","value":"21"},{"label":"At&t","value":"18"},{"label":"BlackBerry","value":"16"},{"label":"Canon","value":"20"},{"label":"Crucial","value":"4"},{"label":"HTC","value":"15"},{"label":"Intel","value":"8"},{"label":"Kenneth Cole","value":"19"},{"label":"Kodak","value":"23"},{"label":"LG","value":"9"},{"label":"Logitech","value":"14"},{"label":"Microsoft","value":"12"},{"label":"Nokia","value":"13"},{"label":"Olympus","value":"22"},{"label":"Samsung","value":"6"},{"label":"Seagate","value":"10"},{"label":"Sony","value":"17"},{"label":"Western Digital","value":"7"}],"is_user_defined":true,"default_frontend_label":"Manufacturer","frontend_labels":null,"backend_type":"int","source_model":"Magento\\Eav\\Model\\Entity\\Attribute\\Source\\Table","default_value":"","is_unique":"0","validation_rules":[]}

### Actual result (*)
1. "manufacturer" data After PUT request above:

{"is_wysiwyg_enabled":false,"is_html_allowed_on_front":false,"used_for_sort_by":false,"is_filterable":true,"is_filterable_in_search":false,"is_used_in_grid":true,"is_visible_in_grid":false,"is_filterable_in_grid":true,"position":0,**"apply_to":["simple"]**,"is_searchable":"1","is_visible_in_advanced_search":"1","is_comparable":"1","is_used_for_promo_rules":"0","is_visible_on_front":"0","used_in_product_listing":"0","is_visible":true,"scope":"global","attribute_id":83,"attribute_code":"manufacturer","frontend_input":"select","entity_type_id":"4","is_required":false,"options":[{"label":" ","value":""},{"label":"AMD","value":"5"},{"label":"Apple","value":"11"},{"label":"Argus","value":"21"},{"label":"At&t","value":"18"},{"label":"BlackBerry","value":"16"},{"label":"Canon","value":"20"},{"label":"Crucial","value":"4"},{"label":"HTC","value":"15"},{"label":"Intel","value":"8"},{"label":"Kenneth Cole","value":"19"},{"label":"Kodak","value":"23"},{"label":"LG","value":"9"},{"label":"Logitech","value":"14"},{"label":"Microsoft","value":"12"},{"label":"Nokia","value":"13"},{"label":"Olympus","value":"22"},{"label":"Samsung","value":"6"},{"label":"Seagate","value":"10"},{"label":"Sony","value":"17"},{"label":"Western Digital","value":"7"}],"is_user_defined":true,"default_frontend_label":"Manufacturer updated","frontend_labels":null,"backend_type":"int","source_model":"Magento\\Eav\\Model\\Entity\\Attribute\\Source\\Table","default_value":"","is_unique":"0","validation_rules":[]}

2. "manufacturer" data After POST request above:

{"is_wysiwyg_enabled":false,"is_html_allowed_on_front":false,"used_for_sort_by":false,"is_filterable":true,"is_filterable_in_search":false,"is_used_in_grid":true,"is_visible_in_grid":false,"is_filterable_in_grid":true,"position":0,**"apply_to":["simple"]**,"is_searchable":"1","is_visible_in_advanced_search":"1","is_comparable":"1","is_used_for_promo_rules":"0","is_visible_on_front":"0","used_in_product_listing":"0","is_visible":true,"scope":"global","attribute_id":83,"attribute_code":"manufacturer","frontend_input":"select","entity_type_id":"4","is_required":false,"options":[{"label":" ","value":""},{"label":"AMD","value":"5"},{"label":"Apple","value":"11"},{"label":"Argus","value":"21"},{"label":"At&t","value":"18"},{"label":"BlackBerry","value":"16"},{"label":"Canon","value":"20"},{"label":"Crucial","value":"4"},{"label":"HTC","value":"15"},{"label":"Intel","value":"8"},{"label":"Kenneth Cole","value":"19"},{"label":"Kodak","value":"23"},{"label":"LG","value":"9"},{"label":"Logitech","value":"14"},{"label":"Microsoft","value":"12"},{"label":"Nokia","value":"13"},{"label":"Olympus","value":"22"},{"label":"Samsung","value":"6"},{"label":"Seagate","value":"10"},{"label":"Sony","value":"17"},{"label":"Western Digital","value":"7"}],"is_user_defined":true,"default_frontend_label":"Manufacturer","frontend_labels":null,"backend_type":"int","source_model":"Magento\\Eav\\Model\\Entity\\Attribute\\Source\\Table","default_value":"","is_unique":"0","validation_rules":[]}

"apply_to" property value remains unchanged after both calls; expected to be amended. "default_frontend_label" property gets amended as appropriate.

**Comment by magento-engcom-team** [comment]
Hi @magolovanov. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@magolovanov do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @milindsingh. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by milindsingh** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @milindsingh. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @milindsingh, here is your [Magento instance](https://i-20629-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-20629-2-3-develop.instances.magento-community.engineering/admin](https://i-20629-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by milindsingh** [comment]
@magento-engcom-team give me 2.2-develop instance

**Comment by magento-engcom-team** [comment]
Hi @milindsingh. Thank you for your request. I'm working on Magento 2.2-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @milindsingh, here is your [Magento instance](https://i-20629-2-2-develop.instances.magento-community.engineering).
Admin access: [https://i-20629-2-2-develop.instances.magento-community.engineering/admin](https://i-20629-2-2-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by magento-engcom-team** [comment]
@milindsingh Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-97940` were created

**Comment by magento-engcom-team** [comment]
Hi @ankur-cedcoss. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by magento-engcom-team** [comment]
Hi @maheshWebkul721. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if you want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by tuyennn** [comment]
I still found it works well on 2.2.6 and 2.2-develop

**Comment by engcom-Alfa** [comment]
Hello @magolovanov !

Thank you for contribution and collaboration!

We are not able to reproduce this issue on the lates `2.3-develop` branch by provided steps.

We are closing this issue due to _branch `2.2-develop` was closed and Magento no longer accepts pull requests and issues to the v2.2 release lines to focus all development efforts on v2.3._
 See [Accepted pull requests and ported code](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#rules) for more details

If you still faced this issue on `2.3` please create a new Issue with all required details according to [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines)
